index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/EsIndexProvider.java

```diff
@@ -0,0 +1,204 @@
+                    + "is trying to connect to cluster '{2}'", getRepositoryName(), clusterName);
+            response = client.admin().cluster()
+                    .prepareHealth().setWaitForGreenStatus().execute().actionGet();
+            count++;
+        } while (response.getStatus() != ClusterHealthStatus.GREEN && count < NUMBER_OF_TRIES);
+
+        if (response.getStatus() != ClusterHealthStatus.GREEN) {
+            this.postShutdown();
+            logger().error("Could not connect to cluster " + response.getStatus());
+            throw new RepositoryException("Could not connect to elasticsearch cluster: " + clusterName);
+        }
+
+        logger().debug("Elasticsearch index provider for repository '{1}' "
+                + "got green light for cluster '{2}'", getRepositoryName(), clusterName);
+    }
+
+    private void disconnect() {
+        try {
+            //close client
+            if (client != null) {
+                client.close();
+            }
+
+            //stop node
+            if (esNode != null) {
+                esNode.stop();
+            }
+        } catch (Exception e) {
+            logger().warn("Can't gracefully shutdow Elasticsearch instance");
+        }
+        client = null;
+        esNode = null;
+    }
+
+    private void connect() {
+        Settings localSettings = ImmutableSettings.settingsBuilder()
+                .put("http.enabled", false)
+                .put("number_of_shards", 1)
+                .put("number_of_replicas", 1).build();
+        //configure Elasticsearch node
+        esNode = isLocalInstance()
+                ? nodeBuilder().settings(localSettings).local(true).build().start()
+                : nodeBuilder().clusterName(clusterName).node();
+
+        //intialize communication
+        client = esNode.client();
+    }
+
+    /**
+     * Gets the name of the Elasticsearch cluster.
+     *
+     * @return the name of the cluster.
+     */
+    public String getClusterName() {
+        return clusterName;
+    }
+
+    private NodeBuilder nodeBuilder() {
+        return NodeBuilder.nodeBuilder();
+    }
+
+    private boolean isLocalInstance() {
+        return clusterName.equalsIgnoreCase(LOCAL);
+    }
+
+    @Override
+    protected void postShutdown() {
+        logger().debug("Shutting down the elasticsearch index provider '{0}' in repository '{1}'", getName(), getRepositoryName());
+        try {
+            client.close();
+            esNode.stop();
+        } catch (Exception e) {
+            logger().warn("Can't gracefully shutdow Elasticsearch instance");
+        }
+    }
+
+    @Override
+    protected int getCostEstimate() {
+        return IndexCostCalculator.Costs.REMOTE;
+    }    
+    
+    @Override
+    public void validateProposedIndex(ExecutionContext context,
+            IndexDefinition defn,
+            NodeTypes.Supplier nodeTypeSupplier,
+            Problems problems) {
+        // first perform some custom validations
```

Comment:
ok; I forgot the base class only provides an empty implementation since the default validations are performed always.